k9mail/src/main/java/com/fsck/k9/provider/DecryptedFileProvider.java

```diff
@@ -73,4 +83,31 @@ private static File getDecryptedTempDirectory(Context context) {
 
         return directory;
     }
+
+    @MainThread // no need to synchronize for receiverRegistered
```

Comment:
That seems wrong. `registerFileCleanupReceiver()` is called from `createProvidedFile()` which is most likely not called on the main thread.